From 4a4b934b6f47e76bcc0913efac7d14dab4650f1d Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 19 May 2022 17:47:15 +0200
Subject: [PATCH 12/12] Revert "sched/alt: Make sure that __set_task_cpu() gets
 called in set_task_cpu()"

This reverts commit 7cde241762140f0dd041bdb53926fbe5a14bfa6b.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 kernel/sched/alt_core.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 00b3144d7..69fb78270 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -1453,13 +1453,11 @@ void set_task_cpu(struct task_struct *p, unsigned int new_cpu)
 
 	WARN_ON_ONCE(is_migration_disabled(p));
 #endif
-
+	if (task_cpu(p) == new_cpu)
+		return;
 	trace_sched_migrate_task(p, new_cpu);
-
-	if (task_cpu(p) != new_cpu) {
-		rseq_migrate(p);
-		perf_event_task_migrate(p);
-	}
+	rseq_migrate(p);
+	perf_event_task_migrate(p);
 
 	__set_task_cpu(p, new_cpu);
 }
-- 
2.36.1.74.g277cf0bc36

